pkg/auth/identityrequest.go

```diff
@@ -14,24 +14,28 @@
 
 package auth
 
-import "encoding/base64"
+import (
+	"encoding/base64"
+
+	discoveryv1alpha1 "github.com/liqotech/liqo/apis/discovery/v1alpha1"
+)
 
 // IdentityRequest is the common interface for Certificate and ServiceAccount identity request.
 type IdentityRequest interface {
-	GetClusterID() string
+	GetClusterIdentity() discoveryv1alpha1.ClusterIdentity
 	GetToken() string
 	GetPath() string
 }
 
 // ServiceAccountIdentityRequest is the request for a new ServiceAccount validation.
 type ServiceAccountIdentityRequest struct {
-	ClusterID string `json:"clusterID"`
-	Token     string `json:"token"`
+	ClusterIdentity discoveryv1alpha1.ClusterIdentity `json:"cluster"`
+	Token           string                            `json:"token"`
 }
 
 // CertificateIdentityRequest is the request for a new certificate validation.
 type CertificateIdentityRequest struct {
-	ClusterID string `json:"clusterID"`
+	ClusterIdentity discoveryv1alpha1.ClusterIdentity `json:"cluster"`
```

Comment:
Is this change required? The certificate should contain the unique clusterID only